### ISSUE TITLE
completion within arrows in php

### ISSUE THREAD

**Issue by heimer16** [hrsh7th/nvim-cmp#879]
### FAQ

- [X] I have checked the [FAQ](https://github.com/hrsh7th/nvim-cmp/blob/15f08a8faa22d52480cdcb9ef9ca698120f04363/doc/cmp.txt#L616) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/hrsh7th/nvim-cmp/issues) and there are no open or closed issues with the same problem.

### Neovim Version

NVIM v0.7.0-dev+1416-g012c05580

### Minimal reproducible full config

if has('vim_starting')
  set encoding=utf-8
endif
scriptencoding utf-8

if &compatible
  set nocompatible
endif

let s:plug_dir = expand('/tmp/plugged/vim-plug')
if !filereadable(s:plug_dir .. '/plug.vim')
  execute printf('!curl -fLo %s/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim', s:plug_dir)
end

execute 'set runtimepath+=' . s:plug_dir
call plug#begin(s:plug_dir)
Plug 'hrsh7th/nvim-cmp'
Plug 'hrsh7th/cmp-buffer'
Plug 'hrsh7th/cmp-nvim-lsp'
Plug 'hrsh7th/vim-vsnip'
Plug 'neovim/nvim-lspconfig'
call plug#end()
PlugInstall | quit

" Setup global configuration. More on configuration below.
lua << EOF
local cmp = require "cmp"
cmp.setup {
  snippet = {
    expand = function(args)
      vim.fn["vsnip#anonymous"](args.body)
    end,
  },

  mapping = {
    ['<CR>'] = cmp.mapping.confirm({ select = true })
  },

  sources = {
    { name = "nvim_lsp" },
    { name = "buffer" },
  },
}
EOF

lua << EOF
local capabilities = require('cmp_nvim_lsp').update_capabilities(vim.lsp.protocol.make_client_capabilities())

require'lspconfig'.cssls.setup {
  capabilities = capabilities,
}
EOF

### Description

completing a word with a following -> in php includes the - in the replacement.

### Steps to reproduce

starting with 
$test->something->whatever
start replacing "something" with "what...."  (i.e. c-w to change word) select from the list and accept completion

### Expected behavior

$test->whatever->whatever
the hyphen should remain after completion

### Actual behavior

$test->whatever>whatever
the hyphen after the completion is "eaten"

### Additional context

|word|_the_rest_of_the_chars
if I replace just "word" and accept completion, _the_rest_of_the_chars gets replaced to.  is there a setting to disable this?

I know there are settings in vim to define what characters determine a new word, but in default settings:
this_is_one_word
these-are-separate-words
so I would think the replacement would not include the hyphen -

**Comment by hrsh7th** [comment]
Your minimal vimrc does not specify php LSP server.

I can't reproduce with intelephense.

**Comment by heimer16** [comment]
My apologies, I have corrected/cleaned up the minimum config.  (apparently nvim -u was still loading other things from my runtime paths, so I completely removed them) 
I added the necessary `behavior = cmp.ConfirmBehavior.Replace`
the only completion is source is "buffer", since this does not involve lsp

with this setup, I still get the same behavior.  the main point I believe is that I'm doing "c-w" to change word on "something" which then places the cursor like this:  `$this->|->another`, which means any confirmation of completion will consider the - as a word, and will get replaced, since the behavior setting is set to `cmp.ConfirmBehavior.Replace`.

now that I understand the behavior a bit better, I can see that this is technically correct behavior, but for me still a bit unexpected with `c-w` for change word.  
I've gone back to using `cmp.ConfirmBehavior.Insert` rather than `cmp.ConfirmBehavior.Replace`
